### PR TITLE
LIZA-134: Clock: Раздельное отслеживание часовой и минутной стрелок

### PR DESCRIPTION
**Pull request by ol00lo**

### CODE REVIEW

File: clock/customs.py
Comment (kalininei) [inline clock/customs.py:1]: image
Как понимать эту метрику?
Вы же докладываете, что там за 50% точность. А тут я вижу 4 только

File: clock/new_train.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import os
+import numpy as np
+import random
+import math
+import tensorflow as tf
+from tensorflow.keras.applications import MobileNet
+from tensorflow.keras import layers, callbacks
+import customs
+
+seed_value = np.random.randint(0, 2**32 - 1)
```
Comment (kalininei) [inline clock/new_train.py:10]: Зачем вы случайную инициализацию делаете?
Мне казалось, что seed фиксируют, чтобы повторяемость обеспечить.

File: clock/new_train.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        for ind in indices:
+            data_file = data_files[ind]
+            label_file = label_files[ind]
+            data_path = os.path.join(data_dir, data_file)
+            labels_path = os.path.join(data_dir, label_file)
+            
+            batch_data = np.load(data_path)
+            batch_labels = np.load(labels_path)
+
+            sum_minutes = batch_labels.astype(np.float32)
```
Comment (kalininei) [inline clock/new_train.py:45]: Вы входные файлы переделываете чтоли?
Разве мы их не зафиксировали?
Чтобы посчитать `sum_minutes` вроде не надо ничего переделывать. Функцию расчёта можно было прям тут написать. Нет?

File: clock/new_train.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+np.random.seed(seed_value)
+
+class SumMinutesLayer(layers.Layer):
+    def __init__(self, **kwargs):
+        super(SumMinutesLayer, self).__init__(**kwargs)
+    
+    def call(self, inputs):
+        hours, minutes = inputs
+        hours = tf.floor(hours)
+        sum_min = hours * 60 + minutes
```
[the remark this answers is not in the document]
Comment (ol00lo) [reply clock/new_train.py:21]: ✔️

File: clock/new_train.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    model = tf.keras.Model(inputs=inp, outputs=[hours, minutes, sum_minutes])
+
+    log_dir = os.path.join("logs", "fit", "model_run")
+
+    model.compile(
+        optimizer=tf.keras.optimizers.Adam(learning_rate=5e-4),
+        loss={
+            'hours_output': customs.hours_loss,
+            'minutes_output': customs.minutes_loss,
+            'sum_minutes': lambda y_true, y_pred: tf.zeros_like(y_true)
```
[the remark this answers is not in the document]
Comment (ol00lo) [reply clock/new_train.py:107]: В ошибке пишет, что ожидает 3 входа, а получает 2

File: clock/new_train.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+class SumMinutesLayer(layers.Layer):
+    def __init__(self, **kwargs):
+        super(SumMinutesLayer, self).__init__(**kwargs)
+    
+    def call(self, inputs):
+        hours, minutes = inputs
+        hours = tf.floor(hours)
+        sum_min = hours * 60 + minutes
+        return tf.stop_gradient(sum_min)
```
[the remark this answers is not in the document]
Comment (ol00lo) [reply clock/new_train.py:22]: Как я понимаю, благодаря этому 3 выход не обучается

File: clock/new_train.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            'minutes_output': 0.5,
+            'sum_minutes': 0.0
+        },
+        metrics={
+            'sum_minutes': [
+                customs.build_accuracy_metrics(10),
+                customs.old_loss
+           ]
+        },
+        jit_compile=False
```
[the remark this answers is not in the document]
Comment (ol00lo) [reply clock/new_train.py]: ✔️

File: clock/customs.py
Comment (ol00lo) [reply clock/customs.py:1]: Я её не понимаю) 
Тут на графике показывается средняя точность по батчам, а мои 50% считаются по всем картинкам без усреднения. Но почему оно в 10 раз меньше получается мне не понятно

File: clock/new_train.py (hunk 1 above)
Comment (ol00lo) [reply clock/new_train.py]: ✔️

File: clock/new_train.py (hunk 2 above)
Comment (ol00lo) [reply clock/new_train.py]: ✔️

File: clock/customs.py
Comment (kalininei) [reply clock/customs.py:1]: ну ок. Оставлять такое без объяснения нельзя конечно, поэтому разберёмся вместе.
